### PR TITLE
Add theseus-ai via grayskull

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
Kind reminder @help-python-c :)

File: recipes/theseus-ai/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  host:
+    - python
+    - pytorch
+    - pip
+  run:
+    - python
+    - numpy
+    - scipy >=1.5.3
+    - scikit-sparse >=0.4.5
+    - pytest >=6.2.1
```
Comment (xhochy) [inline recipes/theseus-ai/meta.yaml:45]: Awkward to see `pytest` as a run dependency. Is this intentional?

File: recipes/theseus-ai/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/theseus-ai-{{ version }}.tar.gz
+  sha256: 150702236782c76815ad349d2089ef801469a9a8073610da77fe4b89a11ad5d1
+
+build:
+  number: {{ build }}
```
Comment (xhochy) [inline recipes/theseus-ai/meta.yaml:20]:
```suggestion
  number: 0
```

Don't weigh down via build number but use the following pattern: https://github.com/conda-forge/tensorflow-feedstock/blob/f73deb53c2b170ca3116b4faee9e58c4409b6717/recipe/meta.yaml#L132

File: recipes/theseus-ai/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+{% set name = "theseus-ai" %}
+{% set version = "0.1.3" %}
+{% set build = 0 %}
+
+{% set processor = "cpu" if cuda_compiler_version == "None" else "gpu" %}  # [not osx]
+{% set processor = "cpu" %}  # [osx]
+
+# Prioritize gpu build if cudatoolkit can be installed (through __cuda virtual package)
+{% set build = build + 100 if processor == "gpu" else build %}
```
Comment (xhochy) [inline recipes/theseus-ai/meta.yaml:9]: See below
```suggestion
```

File: recipes/theseus-ai/meta.yaml (hunk 1 above)
Comment (Tobias-Fischer) [reply recipes/theseus-ai/meta.yaml]: It wasn't - thanks for the catch!

File: recipes/theseus-ai/meta.yaml (hunk 2 above)
Comment (Tobias-Fischer) [reply recipes/theseus-ai/meta.yaml]: Thanks - I hadn't used track_features before :)

**Comment by Tobias-Fischer** [comment]
Thanks again for the review @xhochy - I think this should now be ready :)

**Comment by Tobias-Fischer** [comment]
Kind reminder @conda-forge/help-python-c

**Review by xhochy — APPROVED** [review]

**Comment by Tobias-Fischer** [comment]
Thanks @xhochy!

**Pull request merged**